### PR TITLE
Make sure that moderators can interact with trashcans and restrict paginator usage

### PR DESCRIPTION
**Pull request by Akarys42**
* Add an override for moderation roles in `wait_for_deletion` 5e93396e1655715187d176d1cfdd32aa54616a1a
* Restrict the paginator usage to `ctx.author` by default and add an override for mods c25929cfef967a87fda96c2d0b26dcaf74dc5f77

### CODE REVIEW

File: bot/pagination.py [hunk 1]
```diff
@@ -242,6 +246,9 @@ def event_check(reaction_: discord.Reaction, user_: discord.Member) -> bool:
                         scale_to_size=scale_to_size)
         current_page = 0
 
+        if not restrict_to_user:
+            restrict_to_user = ctx.author
+
         if not lines:
```
Comment (Inheritanc-e) [inline bot/pagination.py:252]: Why not combine these 2 if statements into one using `and`?

**Review by Inheritanc-e — COMMENTED** [review]
Just one small change.

File: bot/pagination.py (hunk 1 above)
Comment (Inheritanc-e) [reply bot/pagination.py:252]: Oh, my bad. 
Didn't see the entire thing.

File: bot/pagination.py [hunk 2]
```diff
@@ -218,10 +222,10 @@ async def paginate(
         def event_check(reaction_: discord.Reaction, user_: discord.Member) -> bool:
             """Make sure that this reaction is what we want to operate on."""
             no_restrictions = (
-                # Pagination is not restricted
-                not restrict_to_user
```
Comment (MarkKoz) [inline bot/pagination.py:222]: This removes support for unrestricted pagination. Now, all pagination must either be restricted to the given user or staff. Was this not a feature we relied upon?

Comment (Akarys42) [reply bot/pagination.py:222]: As far as I can see, defaulting to `ctx.author` won't break anything, the features using the paginator should still work correctly.

File: bot/pagination.py [hunk 3]
```diff
@@ -218,10 +222,10 @@ async def paginate(
...
                 # The reaction was by a whitelisted user
-                or user_.id == restrict_to_user.id
+                user_.id == restrict_to_user.id
+                # The reaction was by a moderator
+                or any(role.id in MODERATION_ROLES for role in user_.roles)
```
Comment (MarkKoz) [inline bot/pagination.py:228]: I don't believe the paginator is inherently restricted to guilds. Therefore, this event may get a `User` which doesn't have `roles`.

**Review by MarkKoz — CHANGES_REQUESTED** [review]

File: bot/pagination.py (hunk 3 above)
Comment (Akarys42) [reply bot/pagination.py]: That's a good point. Fixed in 46ff6949 !

**Review by MarkKoz — APPROVED** [review]

**Review by HassanAbouelela — APPROVED** [review]

**Review by HassanAbouelela — APPROVED** [review]

**Pull request merged**